src/signalr-hub.ts

```diff
@@ -12,6 +12,7 @@ export class SignalRHub {
     private _state$: Subject<string>;
     private _error$: Subject<SignalRError>;
     private _subjects: { [name: string]: Subject<any> };
+    private _primePromise: JQueryPromise<any>;
```

Comment:
Change to use a Subject<any>

```
private _prime$: Subject<any>;
```

Reply:
The reason I chose a promise over a subject here was, that in this case this really is not an event, let alone an event stream, but rather a semaphore signalling that set up of the connection was completed.
In my opinion it is far more readable to be able to use the `await`-keyword here to wait for this semaphore to be set, rather than subscribe to an observable and dispose of the subscription afterwards. Disposal of the subscription would be necessary, as the primePromise lives for the lifetime of the application, and we don't want to get our memory cluttered with a new subscription every time we invoke the `send`-method.